File: beecrowds/ano_bissexto_ou_ano_nao_bissexto.py
#melhor fazer em python

first_output = True 
while True:
    try:
        year = int(input())
        if not year:
            break
        if not first_output:
            print()
        isLeap = False
        if year % 400 == 0 or (year % 4 == 0 and year % 100 != 0):
            isLeap = True
            print("This is leap year.")
        if year % 15 == 0:
            print("This is huluculu festival year.")
        if year % 55 == 0 and isLeap:
            print("This is bulukulu festival year.")
        if (year % 400 != 0 and (year % 4 != 0 or year % 100 == 0)) and year % 15 != 0:
            print("This is an ordinary year.")
        first_output = False  
    except EOFError:
        break

File: beecrowds/ano_bissexto_ou_ano_nao_bissexto.cpp
#include <iostream>

using namespace std;

int main(){
    
    long long int year;
    while(cin >> year){
        bool isLeap = false;
        if(year % 400 == 0 || (year % 4 == 0 && year % 100 != 0)){
            isLeap = true;
            cout << "This is leap year.\n";

        }
        if(year % 15 == 0){
            cout << "This is huluculu festival year.\n";
        }

        if(year % 55 == 0 && isLeap == true){
            cout << "This is bulukulu festival year.\n";
        }

        if((year % 400 != 0 && (year % 4 != 0 || year % 100 == 0)) && year % 15 != 0){
            cout << "This is an ordinary year.\n";
        }
        cout << endl;
    }
}